src/software/amazon/ion/Timestamp.java

```diff
@@ -83,6 +83,18 @@
     private static final int NO_SECONDS = 0;
     private static final BigDecimal NO_FRACTIONAL_SECONDS = null;
 
+    // 0001-01-01T00:00:00.0Z in millis
+    private static final long MINIMUM_ALLOWED_TIMESTAMP_IN_MILLIS = -62135769600000L;
+
+    // 0001-01-01T00:00:00.0Z in millis
+    static final BigDecimal MINIMUM_ALLOWED_FRACTIONAL_MILLIS = new BigDecimal(MINIMUM_ALLOWED_TIMESTAMP_IN_MILLIS);
+
+    // 9999-12-31T23:59:59.999999999999Z in millis
+    static final BigDecimal MAXIMUM_ALLOWED_FRACTIONAL_MILLIS = new BigDecimal("253402300799999.999999999");
+
+    // determined empirically as a safe scale
```

Comment:
For the max millis Timestamp already imposes a max it can represent which is any timestamp before `10000T` as Todd corrected me 

A scale of `10000` for milliseconds means you can have the fractional second to be `1e-10003` which should be enough